server/src/test/java/org/elasticsearch/common/time/TimeSupplierUtils.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.common.time;
+
+import org.elasticsearch.core.TimeValue;
+
+import java.util.function.LongSupplier;
+
+public class TimeSupplierUtils {
+
+    public static TimeSupplier create(LongSupplier timeSourceInMillis) {
+        return new TimeSupplier() {
+            @Override
+            public long timeInMillis() {
+                return timeSourceInMillis.getAsLong();
+            }
+
+            @Override
+            public long timeInNanos() {
+                return timeSourceInMillis.getAsLong() * TimeValue.NSEC_PER_MSEC;
+            }
+        };
+    }
```

Comment:
NIT: It would be nice to have a toString implementation on this one as well since we have multiple implementations that could be used. Also it could be a static method on interface (similar to `List.of(..)`)